src/drivers/usart/usart_m3.c

```diff
@@ -89,10 +89,13 @@ int csp_usart_open(const csp_usart_conf_t *conf, csp_usart_callback_t rx_callbac
 void csp_sciNotification(sciBASE_t *sci, unsigned flags) {
     portBASE_TYPE xHigherPriorityTaskWoken = pdFALSE;
 
+
+
     switch (flags) {
     case SCI_RX_INT:
         xQueueSendToBackFromISR( sciData, &incomingData, &xHigherPriorityTaskWoken );
         sciReceive(sci, sizeof(uint8_t), &incomingData);
+        uhf_pipe_timer_reset_from_isr(&xHigherPriorityTaskWoken);

```

Comment:
So the timer resets when something is received?

Reply:
Yes to align with how the UHF radio works. Whenever it sends or receives data, the timeout resets